pkgs/by-name/kc/kcc/package.nix

```
@@ -0,0 +1,62 @@
+{
+  stdenv,
+  lib,
+  qt6,
+  python3Packages,
+  fetchFromGitHub,
+  p7zip,
+  archiveSupport ? true,
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "kcc";
+  version = "6.2.2";
+
+  src = fetchFromGitHub {
+    owner = "ciromattia";
+    repo = "kcc";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-GX0WPu1EZx/8H1ajivxgnym4jECmwPBJU25vzcfOQCw=";
+  };
+
+  nativeBuildInputs = [ qt6.wrapQtAppsHook ];
+
+  buildInputs = lib.optional stdenv.hostPlatform.isLinux qt6.qtwayland;
+  propagatedBuildInputs = (
+    with python3Packages;
+    [
+      packaging
+      pillow
+      psutil
+      python-slugify
+      raven
+      requests
+      natsort
+      mozjpeg_lossless_optimization
+      distro
+      pyside6
+    ]
+  );
+
+  qtWrapperArgs = lib.optionals archiveSupport [
+    "--prefix"
+    "PATH"
+    ":"
+    "${lib.makeBinPath [ p7zip ]}"
+  ];
+
+  postFixup = ''
+    wrapProgram $out/bin/kcc "''${qtWrapperArgs[@]}"
+  '';
+
```

I think you can add in the `passthru.updateScript` section to enable automatic updates.

```suggestion
  passthru = {
    updateScript = nix-update-script { };
  };
```